packages/session-manager/src/index.tsx

```diff
@@ -0,0 +1,95 @@
+import { user as userAPI } from '@tryrolljs/api'
+import Client from '@tryrolljs/api-client'
+import { auth } from '@tryrolljs/sdk'
+import {
+  useState,
+  useEffect,
+  useCallback,
+  cloneElement,
+  createContext,
+  PropsWithChildren,
+  ReactElement,
+  useContext,
+} from 'react'
+
+export const SessionContext = createContext<{
+  user?: userAPI.types.GetMeResponseData
+  logIn: () => void
+  logOut: () => Promise<void>
+  error?: unknown
+}>({
+  logIn: () => {},
+  logOut: () => Promise.resolve(),
+})
+
+type Props = PropsWithChildren<{
+  apiClient: Client
+  authSdk: auth.AuthSDK
+}>
+
+const OAUTH_CODE_URL_PARAM_KEY = 'code'
+
+const SessionProvider = ({ apiClient, authSdk, children }: Props) => {
+  const [user, setUser] = useState<userAPI.types.GetMeResponseData>()
+  const [error, setError] = useState<unknown>()
+
+  useEffect(() => {
+    const loadUserData = async () => {
+      try {
+        const user_ = await userAPI.getMe(apiClient)
+        setUser(user_)
+      } catch (e) {
+        setError(e)
+      }
+    }
+
+    const getOauthCode = () => {
+      const urlParams = new URLSearchParams(window.location.search)
+      return urlParams.get(OAUTH_CODE_URL_PARAM_KEY)
+    }
+
+    const initializeNewSession = async () => {
```

Comment:
can we ensure that we only try to grab the `code` if we are also on the oauth redirect url? I want to avoid clashing with a `code` param used for something else in a different application.

We will only supply this oauth code as part of that final redirect